shards/core/type_info.cpp

```diff
@@ -131,20 +133,38 @@ SHTypeInfo deriveTypeInfo(const SHVar &value, const SHInstanceData &data, std::v
     if (expInfo) {
       auto sv = SHSTRVIEW(value);
       const auto varName = sv;
-      for (auto info : data.shared) {
-        std::string_view name(info.name);
-        if (name == sv) {
-          expInfo->push_back(SHExposedTypeInfo{.name = info.name, .exposedType = info.exposedType});
+      if (data.composeVersion == 2) {
```

Comment:
Shouldn't this also be allowed on compose v1, since it should still have the privateContext, just filled shared, etc.

Reply:
true